src/test/ui/rfc-2093-infer-outlives/check-predicates.rs

```diff
@@ -0,0 +1,23 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![feature(rustc_attrs)]
+
+// Needs an explicit where clause stating outlives condition. (RFC 2093)
+
+
+#[rustc_outlives]
+struct Foo<'a, T> { //~ ERROR 17:1: 20:2: [Binder(OutlivesPredicate(T, ReEarlyBound(0, 'a)))]
```

Comment:
Nit: you might prefer to have the "error message" be something very simple, like "rustc_outlives", and add the details via calls to `note`. this way, the `//~ ERROR` can be short and sweet, and the details are in the stderr file